frontend/js/src/explore/artist-similarity/artist-search/SearchBox.tsx

```diff
@@ -0,0 +1,181 @@
+    // Lookup the artist based on the query
+    const getArtists = async (): Promise<void> => {
+        if(searchQuery.length && searchQuery.trim() !== ""){
+            let results = await artistLookup(searchQuery);
+            setSearchResults(results ?? []);
+            //console.log(searchResults);
+            // Open the dropdown if any results exist
+            if(searchResults.length) {
+                //console.log(searchResults);
+                setOpenDropdown(true);
+            }
+        }
+        else{
+            setSearchResults([]);
+        }
+    };
+
+    const handleQueryChange = (query: string) => {
+        setSearchQuery(query);
+    }
+
+    useEffect(() => {
+        getArtists();
+    },[searchQuery]);
+    */
+
+  /**
+   * Solution:
+   *  I was able to look up an article for the problem, https://www.developerway.com/posts/debouncing-in-react
+   *  The solution works as expected but I feel like it might be a bit hard to understand what's going on.
+   */
+
+  const getArtists = useCallback(async () => {
+    if (searchQuery.length && searchQuery.trim().length) {
+      const results = await artistLookup(searchQuery);
+      setSearchResults(results);
+      setOpenDropdown(true);
+    }
+  }, [searchQuery]);
+  const getArtistsRef = useRef(getArtists);
+
+  useEffect(() => {
+    // Update the getArtists reference when searchQuery gets updated.
+    getArtistsRef.current = getArtists;
+  }, [getArtists, searchQuery]);
+
+  // Create a throttled function using the current reference of getArtists
+  const throttledGetArtists = useMemo(() => {
+    return throttle(
+      () => {
+        getArtistsRef.current?.();
+      },
+      800,
+      { leading: false, trailing: true }
+    );
+  }, []);
+
+  // Lookup the artist based on the query
+  const handleQueryChange = (query: string) => {
+    setSearchQuery(query);
+    throttledGetArtists();
+  };
+
+  const increment = () => {
+    onSimilarArtistsLimitChange(currentsimilarArtistsLimit + 1);
+  };
+  const decrement = () => {
+    onSimilarArtistsLimitChange(currentsimilarArtistsLimit - 1);
+  };
+  return (
+    <form className="user-inputs-container" autoComplete="off">
+      <div
+        className="artist-input-container"
+        onFocus={() => searchResults.length && setOpenDropdown(true)}
+        onBlur={() => setTimeout(() => setOpenDropdown(false), 100)}
+      >
+        <div className="searchbox-container">
+          <input
+            id="searchbox-artist-name"
+            type="search"
+            name="artist_mbid"
+            placeholder="Artist name"
+            onChange={(e) => handleQueryChange(e.target.value)}
+            value={searchQuery}
+          />
+          <button id="searchbox-icon" type="button">
+            <FontAwesomeIcon icon={faMagnifyingGlass} color="white" />
+          </button>
+        </div>
+        <div className="search-dropdown-container">
+          {openDropdown && (
+            <SearchDropdown
```

Comment:
`SearchDropdown` component seems simple enough that you can directly inline the code here. There are no other uses of the component AFAICS and it does not maintain any state either.

Reply:
Fixed